app/javascript/components/greetings.jsx

```diff
@@ -0,0 +1,22 @@
+import React, {useEffect} from 'react'
+    import { useDispatch, useSelector } from "react-redux";
+    import { getAllGreets, fetchGreetingAsyncThank } from "../redux/greetReducer";
+    const Greeting = () => {
+        const greetList = useSelector(getAllGreets);
+        console.log(greetList)
+        const dispatch = useDispatch();
+        useEffect(() => {
+            if (greetList.length === 0 )
+                dispatch(fetchGreetingAsyncThank());
+        }, [dispatch]);
+
+        return <ul>
+            Top {greetList.length} greetings
+            {greetList.map((greet) => (
+                    <li>{greet.message}</li>
+                )
+            )}
+        </ul>;
```

Comment:
- [ ] You really did a great job but please as part of the requirements, you are supposed to display one random greeting from the list and not all of them at once
<img width="1017" alt="image" src="https://user-images.githubusercontent.com/37901469/223634431-8a9410d6-8018-4f8f-b55c-ca362e461a4f.png">

Reply:
Thanks for your constructive feedback.